doc/specs/stdlib_stats_distribution_exponential.md

```diff
@@ -0,0 +1,243 @@
+---
+title: stats_distribution_exponential
+---
+
+# Statistical Distributions -- Exponential Distribution Module
+
+[TOC]
+
+## `rvs_expon` - exponential distribution random variates
+
+### Status
+
+Experimental
+
+### Description
+
+An exponentially distributed random variate distribution is the distribution of time between events in a Poisson point process. The inverse scale parameter `lambda` specifies the rate of change.
+
+Without argument the function returns a standard exponential distributed random variate E(1) with `lambda = 1`.
+
+With single argument, the function returns an exponential distributed random variate E(lambda). For complex arguments, the real and imaginary parts are independent of each other.
+
+With two arguments the function returns a rank one array of exponential distributed random variates.
+
+Note: the algorithm used for generating normal random variates is fundamentally limited to double precision.
+
+### Syntax
+
+`result = [[stdlib_stats_distribution_exponential(module):rvs_expon(interface)]]([lambda] [[, array_size]])`
+
+### Class
+
+Function
+
+### Arguments
+
+`lambda`: optional argument has `intent(in)` and is a scalar of type `real` or `complex`.
```

Comment:
I suppose `lambda` has to be nonnegative (and restricted to one quadrant for the complex case). Perhaps worth mentioning that here.